packages/web-react/src/components/TextFieldBase/useTextFieldBase.ts

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+import { TextFieldType } from '../../types';
+
+export interface UseTextFieldBaseProps {
+  hasPasswordToggle?: boolean;
+  isMultiline?: boolean;
+  type?: TextFieldType;
+}
+
+export interface UseTextFieldBaseReturn {
+  inputType?: TextFieldType;
+  passwordShown: boolean;
+  passwordToggleHandler: () => void;
+}
+
+export const useTextFieldBase = (props: UseTextFieldBaseProps): UseTextFieldBaseReturn => {
+  const { hasPasswordToggle, isMultiline, type } = props;
+  const [passwordShown, setPasswordShown] = useState<boolean>(false);
+
+  let inputType = type;
+  if (isMultiline) {
+    inputType = undefined;
+  } else if (hasPasswordToggle) {
+    inputType = passwordShown ? 'text' : 'password';
+  }
```

Comment:
This could be a function. But I'm not the one to tell what's right, @literat is.

Reply:
Refactored.